packages/happy-dom/src/xml-parser/XMLParser.ts

```diff
@@ -24,7 +24,7 @@ import * as Entities from 'entities';
  * Group 8: End of start tag (e.g. ">" in "<div>").
  */
 const MARKUP_REGEXP =
-	/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([^-]+)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm;
+	/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([^-]+(?:-[^-]+)*)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm;
```

Comment:
```suggestion
	/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([\s\S]+)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm;
```

Reply:
My suggestion is that we just capture all characters until "-->" instead.

Reply:
I have commited my suggestion now as I would like to release a fix as soon as possible :sweat_smile:

Reply:
That suggestion did not work. The problem is that the RegExp you suggested doesn't cover when there multiple dashes (e.g. "<!-- a-b-c -->").

I'm looking into another solution.

Reply:
I think making it non-greedy should work so 

`/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([\s\S]+?)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm`